autoload/dirvish.vim

```diff
@@ -388,6 +388,7 @@ function! dirvish#open(...) range abort
     return
   endif
   if !&hidden && &modified
+      \ && (!exists("*win_findbuf") || len(win_findbuf(winbufnr(0))) == 1)
```

Comment:
`win_findbuf` returns a list of windows containing that that buffer.  It should be at least 1, for the current window that we got `winbufnr(0)` from.  If the length is exactly one, then opening something else in this window would lose the modified buffer.  Otherwise, the other windows will keep the modified buffer alive while dirvish changes this window.